app/src/main/java/com/alexP/assignment1/ui/addContactFragment/AddContactFragment.kt

```diff
@@ -0,0 +1,149 @@
+import com.alexP.assignment1.utils.getValidationResultMessage
+import com.alexP.assignment1.utils.loadCircularImage
+import com.alexp.contactsprovider.data.Contact
+import com.alexp.textvalidation.data.validator.EmailValidator
+import com.alexp.textvalidation.data.validator.EmptyValidator
+import com.alexp.textvalidation.data.validator.base.BaseValidator
+import com.alexp.textvalidation.data.validator.base.ValidationResult
+import kotlinx.coroutines.launch
+
+
+class AddContactFragment(
+    private val onSaveAction: (Contact) -> Unit
+) : DialogFragment() {
+
+    private lateinit var binding: FragmentDialogAddContactBinding
+    private lateinit var viewModel: AddContactViewModel
+
+    private val galleryLauncher =
+        registerForActivityResult(ActivityResultContracts.GetContent()) { uri ->
+            if (uri == null) return@registerForActivityResult
+            try {
+                viewModel.setGalleryUri(uri)
+                binding.imageViewProfileImage.loadCircularImage(uri.toString())
+            } catch (e: Exception) {
+                e.printStackTrace()
+            }
+        }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentDialogAddContactBinding.inflate(inflater, container, false)
+        binding.root.applyWindowInsets()
+
+        viewModel = ViewModelProvider(this)[AddContactViewModel::class.java]
+
+        observeValues()
+        val toolbar = binding.topBar
+        toolbar.setNavigationOnClickListener {
+            dismiss()
+        }
+        setListeners()
+
+        return binding.root
+    }
+
+    private fun observeValues() {
+        lifecycleScope.launch {
+            viewModel.galleryUri.collect { uri ->
+                uri?.let {
+                    binding.imageViewProfileImage.loadCircularImage(it.toString())
+                }
+            }
+        }
+    }
+
+    private fun setListeners() {
+        binding.buttonSave.setOnClickListener {
+            onButtonSavePressed()
+        }
+        binding.imageViewAddProfileImage.setOnClickListener {
+            galleryLauncher.launch("image/*")
+        }
+    }
+
+    private fun onButtonSavePressed() {
+        if (enteredDataIsInvalid()) return
+
+        val contact = Contact(
+            id = -1,
+            photo = viewModel.galleryUri.toString(),
+            fullName = binding.inputEditTextUsername.text.toString(),
+            career = binding.inputEditTextCareer.text.toString(),
+            email = binding.inputEditTextEmail.text.toString(),
+            phone = binding.inputEditTextPhone.text.toString(),
+            address = binding.inputEditTextAddress.text.toString(),
+            dateOfBirth = binding.inputEditTextDateOfBirth.text.toString()
+        )
+        onSaveAction(contact)
+        dismiss()
+    }
+
+    private fun enteredDataIsInvalid(): Boolean {
```

Comment:
This function is performing asserting inputted string is valid. Things that are not related to UI layer have to be extracted to other layers and be calling from viewmodel.